[anthonynmh] iP

anthonynmh opened the pull request:
# DukePro - Jarvis

   >“Your mind is for having ideas, not holding them.” – David Allen \([source](https://dansilvestre.com/productivity-quotes))
   
DukePro frees your mind of having to remember things you need to do. It's,

- text-based
* easy to learn
+ ~FAST~ _SUPER_ FAST to use 

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```


<!--
This is a project template for a greenfield Java project. It's named after the Java mascot _Duke_. Given below are instructions on how to use it.

## Setting up in Intellij

Prerequisites: JDK 11, update Intellij to the most recent version.

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)
1. Open the project into Intellij as follows:
   1. Click `Open`.
   1. Select the project directory, and click `OK`.
   1. If there are any further prompts, accept the defaults.
1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
   In the same dialog, set the **Project language level** field to the `SDK default` option.
3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
-->
   ```
   Hello from
    ____        _        
   |  _ \ _   _| | _____ 
   | | | | | | | |/ / _ \
   | |_| | |_| |   <  __/
   |____/ \__,_|_|\_\___|
   ```

Commit:
Add Gradle support

anthonynmh committed:
Done with skeleton code for Level-1

anthonynmh committed:
Added functionality for todo and list

anthonynmh committed:
Added ToDoList class to handle data of todo tasks

anthonynmh committed:
Updated with NumberFormatException handling

anthonynmh committed:
Added the DukeException file

anthonynmh committed:
Created a Task class

anthonynmh committed:
Updated array to store Task objects instead of String

anthonynmh committed:
Completed Level-4

anthonynmh committed:
Updated Jarvis to handle custom exceptions

anthonynmh committed:
Created custom exception

anthonynmh committed:
Bug fixes

anthonynmh committed:
Updated name to match Jarvis

anthonynmh committed:
Updated input file

anthonynmh committed:
Updated expected file

anthonynmh committed:
Level-6

anthonynmh committed:
Add jarvis.txt

anthonynmh committed:
Update the display format

anthonynmh committed:
Update the display format

anthonynmh committed:
Update printTask method

anthonynmh committed:
Update variable name

anthonynmh committed:
Update to handle saving and loading tasks

anthonynmh committed:
Add class to handle file

anthonynmh committed:
Update constructors to accept LocalDate

anthonynmh committed:
Update constructors to accept LocalDate

anthonynmh committed:
Merge branch 'branch-Level-7'

Jarvis does not support reading and writing to and from memory - text file.

Tasks in ToDoList is unsaved after 'bye' command.

Add FileHandler to handle read and write to and from memory - text file.

Task variables being String helps in storing using FileHandler.

anthonynmh committed:
Update to merge with branch-Level-8

anthonynmh committed:
Update FileHandler to support LocalDate

anthonynmh committed:
Make more OOP

anthonynmh committed:
Move all classes into a package

anthonynmh committed:
Merge branch 'add-gradle-support' of github.com:anthonynmh/ip

add-gradle-support

anthonynmh committed:
Add gradle support

anthonynmh committed:
Update JUnit dependency in gradle

anthonynmh committed:
Add test files

anthonynmh committed:
Remove extra line space

anthonynmh committed:
Make Jar

anthonynmh committed:
Add JavaDocs

anthonynmh committed:
Add JavaDocs

anthonynmh committed:
Update code to standards

anthonynmh committed:
Update saved list

anthonynmh committed:
Add find functionality for tasks

anthonynmh committed:
Merge branch 'branch-A-JavaDoc'

Update JavaDocs for all methods

anthonynmh committed:
Merge branch 'branch-A-CodingStandard'

Change code to follow course standards

anthonynmh committed:
Merge branch 'branch-Level-9'

Add find functionality for finding tasks

venuslimm commented on src/main/java/seedu/duke/Todo.java:
```diff
@@ -0,0 +1,20 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public String printTask() {
```
I think the method name is slightly misleading. Would getTaskStatus be a better name?
```suggestion
    public String getTaskStatus() {
```

anthonynmh replied:
Good point, thank you for the input!

venuslimm commented on src/main/java/seedu/duke/ToDoList.java:
```diff
@@ -0,0 +1,150 @@
+     *
+     * @param index index of task
+     */
+    public void markDone (int index) {
```
Should the extra space be removed?
```suggestion
    public void markDone(int index) {
```

anthonynmh replied:
Good catch!

venuslimm commented on src/main/java/seedu/duke/Task.java:
```diff
@@ -0,0 +1,79 @@
+
+    private int id;
+    private String task;
+    private boolean done;
```
Should this be renamed to `isDone`?
```suggestion
    private boolean isDone;
```

anthonynmh replied:
I agree, `isDone` makes more sense in this case. Thank you for the suggestion!

venuslimm commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,164 @@
+     * @param storage the storage associated with the current ToDoList
+     * @return isBye boolean indicated the ending 'bye' command
+     */
+    public boolean parse(String command, ToDoList todolist, Storage storage) {
```
I have noticed that you named your `ToDoList` objects as `todolist` in this line and other lines. Should the `todolist` variable rename to `toDoList`?
```suggestion
    public boolean parse(String command, ToDoList toDoList, Storage storage) {
```

anthonynmh replied:
I believe that this depends on whether `todolist` is itself a Noun. However, I can see why it may cause confusion. Good catch!

venuslimm reviewed (commented):
LGTM! I have left some suggestions but generally I think your code is easy to read. I like how you separate the different logical units using lines.

anthonynmh committed:
Update README.md

anthonynmh committed:
Add checkstyle

anthonynmh committed:
Add tutorial JavaFX files

anthonynmh committed:
Update checkstyle in master

anthonynmh committed:
Add GUI

anthonynmh committed:
Merge branch 'branch-Level-10'

Add GUI

anthonynmh committed:
Merge branch 'master' of github.com:anthonynmh/ip

CheckStyle

anthonynmh committed:
Update data log

anthonynmh committed:
Update data log

anthonynmh committed:
parse method in Parser class: Abstract methods to follow SLAP

Parse method is too long, it does not follow SLAP.

Long methods causes confusion and not friendly looking.

Abstracting the specifics helps to improve the level of
complexity within the method.

Let's abstract the specific details to enhance SLAP of the parse method.
It will become more friendlier and less complex.

anthonynmh committed:
Sync data

anthonynmh committed:
Add Java assertions

Code do not have Java assertions.

Java assertions can help to ensure the correctness of code.

Add Java assertions to ensure code correctness.

Let's use Java assertions liberally to ensure code correctness
for now, and especially for future developments.

anthonynmh committed:
Improve code quality

anthonynmh committed:
Merge pull request #2 from anthonynmh/branch-A-Assertions

Add Java assertions

anthonynmh committed:
Merge branch 'master' into branch-A-CodeQuality

anthonynmh committed:
Merge pull request #3 from anthonynmh/branch-A-CodeQuality

Improve code quality

anthonynmh committed:
Update to verify date is current

anthonynmh committed:
Add feature to find free time

anthonynmh committed:
Update code to create new data file on new launch

anthonynmh committed:
Add Ui.png

anthonynmh committed:
Change stage title to Jarvis

anthonynmh committed:
Update UG

anthonynmh committed:
Update UG

anthonynmh committed:
Update UG

anthonynmh committed:
Fix JavaFX bug

Reorganised files into proper packages for ease of developing
